common/src/nostr/mod.rs

```diff
@@ -118,12 +118,10 @@ impl NostrClient {
                 }
             }
         };
-        let result = match timeout(Duration::from_secs(timeout_secs), loop_future).await {
+        match timeout(Duration::from_secs(timeout_secs), loop_future).await {
             Ok(result) => result,
             Err(e) => Err(anyhow!("Timeout, {}", e)),
-        };
-
-        result
+        }
```

Comment:
returning the result of a `let` binding from a block
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_and_return
`#[warn(clippy::let_and_return)]` on by defaultclippy[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fworkspaces%2Fcashu-escrow-kit%2Fcommon%2Fsrc%2Fnostr%2Fmod.rs)
mod.rs(117, 9): unnecessary `let` binding
mod.rs(117, 9): return the expression directly: `match timeout(Duration::from_secs(timeout_secs), loop_future).await {
Ok(result) => result,
Err(e) => Err(anyhow!("Timeout, {}", e)),
}`